src/qtt/measurements/scans.py

```diff
@@ -1948,6 +1946,21 @@ def acquire_segments(station, parameters, average=True, mV_range=2000,
 
     return alldata
 
+def _is_m4i(instrument):
+    try:
```

Comment:
why the try clause. Isinstance is just returning True or False, only when qcodes.instrument_drivers.Spectrum.M4i.M4i is not a class you can expect a TypeError